src/user.rs

```diff
@@ -310,27 +310,65 @@ impl User<Synching> {
         !self.state.available_room_ids.is_empty()
     }
 
-    pub async fn create_room(&mut self, friendship: &Friendship) -> Option<Box<RoomId>> {
+    pub async fn create_room(
+        &mut self,
+        friendship: &Friendship,
+        retry_attempts: usize,
+    ) -> Option<Box<RoomId>> {
         let client = self.client.lock().await;
-
-        let instant = Instant::now();
         let alias = friendship.local_part.to_string();
-        let request = assign!(CreateRoomRequest::new(), { room_alias_name: Some(&alias) });
-        let response = client.create_room(request).await;
 
-        match response {
-            Ok(ref response) => {
-                self.send(Event::RequestDuration((
-                    UserRequest::CreateRoom,
-                    instant.elapsed(),
-                )))
-                .await;
-                Some(response.room_id.clone())
+        let mut i = 0;
+
+        loop {
+            if i == retry_attempts {
+                log::warn!(
+                    "Creating room {} failed due to max retry attempts",
+                    friendship.local_part
+                );
+                break None;
             }
-            Err(e) => {
-                println!("Failed to create room {}", e);
-                self.send(Event::Error((UserRequest::CreateRoom, e))).await;
-                None
+
+            i += 1;
+
+            let instant = Instant::now();
+            let request = assign!(CreateRoomRequest::new(), { room_alias_name: Some(&alias) });
+
+            let response = client.create_room(request).await;
+
+            match response {
+                Ok(ref response) => {
+                    self.send(Event::RequestDuration((
+                        UserRequest::CreateRoom,
+                        instant.elapsed(),
+                    )))
+                    .await;
+                    break Some(response.room_id.clone());
+                }
+                Err(e) => {
+                    log::error!("Failed to create room {}", e);
+
+                    if let Some(error_response) = e.uiaa_response() {
+                        if let Some(error_kind) = &error_response.auth_error {
+                            match &error_kind.kind {
+                                ErrorKind::RoomInUse => {
+                                    self.send(Event::Error((UserRequest::CreateRoom, e))).await;
+
+                                    break None;
+                                }
+                                ErrorKind::ResourceLimitExceeded { admin_contact: _ } => {
+                                    self.send(Event::Error((UserRequest::CreateRoom, e))).await;
+
+                                    // TODO! should we have some kind of thread sleep here?
+                                    break None;
```

Comment:
Let's document why in these particular cases we wouldn't be able to retry the room creation.